pkg/session/txn.go

```diff
@@ -70,6 +70,9 @@ type LazyTxn struct {
 
 	// mark the txn enables lazy uniqueness check in pessimistic transactions.
 	lazyUniquenessCheckEnabled bool
+
+	// commit ts of the last successful transaction, to ensure ordering of TS
+	lastCommitTS uint64
```

Comment:
This field seems to overlap with the newly added `CommitTS` in the KV interface. Additionally, `LazyTxn` is a structure that is no longer needed once a transaction is completed. Or just retain only one of them?

Reply:
This is because `kv.Transaction` will be set to nil in `changeToInvalid` in `LazyTxn.Commit`.

Reply:
How about returning the `commitTS` of before `changeToInvalid`, like
```
- func (s *session) doCommit(ctx context.Context) (error) {
+ func (s *session) doCommit(ctx context.Context) (uint64, error) {
	if !s.txn.Valid() {
		return nil
	}

	defer func() {
        // Return the commitTS if error is nil
		s.txn.changeToInvalid()
		s.sessionVars.SetInTxn(false)
		s.sessionVars.ClearDiskFullOpt()
	}()
   ...
}
```

Reply:
Actually the reason is `CommitTS` has to be saved until the next transaction. If we change `doCommit` to return `CommitTS`, we have to,
- save `CommitTS` in `session`
- check if the transaction is committed successfully in `doCommit`
Since we have to save to save `CommitTS`, it would be simpler to save `CommitTS` inside `doCommit`.

Reply:
Can we store commit ts in `doCommit`?

```diff
+func (s *session) doCommit(ctx context.Context) (err error) {
	if !s.txn.Valid() {
		return nil
	}

	defer func() {
+		if err != nil {
+			s.sessionVars.LastCommitTS = s.txn.CommitTS()
+		}
		s.txn.changeToInvalid()
		s.sessionVars.SetInTxn(false)
		s.sessionVars.ClearDiskFullOpt()
	}()
	...
}
```